source/commands/bounty.js

```diff
@@ -0,0 +1,65 @@
+const { PermissionFlagsBits, ActionRowBuilder, StringSelectMenuBuilder } = require('discord.js');
+const { CommandWrapper } = require('../classes');
+const { database } = require('../../database');
+const { getNumberEmoji } = require('../helpers');
+
+const customId = "bounty";
+const options = [];
+const subcommands = [
+	{
+		name: "post",
+		description: "Post your own bounty (+1 XP)",
+		optionsInput: []
+	}
+];
+module.exports = new CommandWrapper(customId, "Bounties are user-created objectives for other server members to complete", PermissionFlagsBits.ViewChannel, false, true, 3000, options, subcommands,
+	(interaction) => {
+		switch (interaction.options.getSubcommand()) {
+			case subcommands[0].name: // Post
+				database.models.Hunter.findOne({ where: { userId: interaction.user.id, guildId: interaction.guildId } }).then(async hunter => {
```

Comment:
This might benefit from a similar await scheme as the code contained within it. The anonymous function will need to be `async` then, but that should be as easy as prepending it with that keyword.

Reply:
I tend to default to .then and only use async/await if it simplifies flow to try and get away with not promisifying things unnecessarily.

Reply:
Well, in this case, it'll keep a lot of the same checks/comparisons in the same flow and should reduce indentation in a meaningful way. We know that this whole function needs to become asynchronous anyway for this check, so there's no harm in having it be async from the get go.

Reply:
Yeah, I guess we probably can't get away with pretending we'll always acknowledge interactions in 3 seconds forever (and thus will have to use a deferred reply). At that point I'd prefer the execute signature to be `(interaction) => Promise<void>` than include the case where it's not returning a promise. So I guess something to note for when we do that class refactor.